jwst/resample/resample_step.py

```diff
@@ -114,7 +116,14 @@ def process(self, input):
             util.update_s_region_imaging(model)
             model.meta.asn.pool_name = input_models.asn_pool_name
             model.meta.asn.table_name = input_models.asn_table_name
-            model.meta.resample.pixel_scale_ratio = self.pixel_scale_ratio
+
+            # if pixel_scale exists, it will override pixel_scale_ratio.
+            # calculate the actual value of pixel_scale_ratio based on pixel_scale
+            # because source_catalog uses this value from the header.
+            if not self.pixel_scale:
```

Comment:
The default value of `pixel_scale` set in the "spec" block is None. Does this evaluate properly, or should it be 
`if self.pixel_scale is None`?